src/core/dexes/BasicDex.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+// imported contracts and libraries
+import "openzeppelin/token/ERC20/IERC20.sol";
+
+// libraries
+import "../../libraries/Errors.sol";
+
+abstract contract BasicDex {
+    modifier afterSwapCheck(address _sellToken, address _buyToken) {
```

Comment:
I don't see the modifier being used anywhere. Is that correct?

Reply:
Yes, thank you for the catch. I've added them to the actual DEX. I was only testing it in the mockDex.